autoconf/m4/ax_check_compiler.m4

```diff
@@ -31,8 +31,16 @@ AC_DEFUN([_AX_CHECK_COMPILER_OPTION_WITH_VAR],
     AC_MSG_CHECKING([whether the _AC_LANG compiler understands $3])
     SAVE_[]_AC_LANG_PREFIX[]FLAGS=${_AC_LANG_PREFIX[]FLAGS}
     SAVE_$2=${$2}
-    _AC_LANG_PREFIX[]FLAGS=$3
+    AS_LITERAL_IF([$3],
```

Comment:
This newly-added code is very GCC-/clang-specific. Ideally, the client would test the non-"no-"-decorated option and then push the "no-"-decorated option to the list of flags on a successful test when the compiler is known to be GCC- or GCC-compatible (such as, clang).

Failing that, this would otherwise have to detect whether the compiler was/is GCC or GCC-compatible and then effect the modifications you propose, if so.

Reply:
Hi @gerickson, how do you propose we fix this? Do you have any recommendation on how to detect if the compiler is GCC or GCC-compatible?

Reply:
See here for an example: https://github.com/nestlabs/nlbuild-autotools/blob/master/autoconf/m4/nl_enable_coverage.m4#L89. That said, I still prefer that this get fixed at the integrating-package configure.ac level. Fundamentally, I don't think this macro should not be doing this kind of hoop-jumping.

Reply:
Ok, I'll close the PR.